src/components/mobile-nav-item/types.ts

```diff
@@ -0,0 +1,21 @@
+import { ReactNode } from 'react'
+
+type CommonMobileNavItemProps = {
+  label: string
+  isActive?: boolean
+  hasBadge?: boolean
+}
+
+export interface MobileNavItemSimpleProps extends CommonMobileNavItemProps {
+  children?: never
+  isDefaultExpanded?: never
+  href: string
+}
+
+export interface MobileNavItemExpandableProps extends CommonMobileNavItemProps {
+  children: ReactNode
+  isDefaultExpanded?: boolean
+  href?: never
+}
```

Comment:
**question:** Do we really need this separation between "simple" and "expandable" props for the component? Also, are we absolutely sure the parent item can't have a URL associated with it?

Reply:
> Do we really need this separation between "simple" and "expandable" props for the component?

Yes, this should be separated since the "expandable" and "simple" variants are actually different in terms of rendering node

simple is equal with `HTMLAnchorElement`, and expandable is equal with `HTMLButtonElement`; and I have confirm this with Andrei that the parent of expandable is only have the expand and collapse responsobility

Reply:
Okay great. If we don't want an href to be valid on the the item when it has children, what you've got it what we need 👍